src/V3Randomize.cpp

```diff
@@ -724,6 +724,45 @@ class ConstraintExprVisitor final : public VNVisitor {
 
         editSMT(nodep, nodep->fromp(), lsbp, msbp);
     }
+    void visit(AstAssocSel* nodep) override {
+        if (editFormat(nodep)) return;
+        FileLine* const fl = nodep->fileline();
+        if (VN_IS(nodep->bitp(), CvtPackString)) {
+            // Extract and truncate the string index to fit within 64 bits
+            AstCvtPackString* const stringp = VN_AS(nodep->bitp(), CvtPackString);
+            VNRelinker handle;
+            AstNodeExpr* const strIdxp = new AstSFormatF{
+                fl, "#x%16x", false,
+                new AstAnd{fl, stringp->lhsp()->unlinkFrBack(&handle),
+                           new AstConst(fl, AstConst::Unsized64{}, 0xFFFFFFFFFFFFFFFF)}};
+            handle.relink(strIdxp);
+            editSMT(nodep, nodep->fromp(), strIdxp);
+        } else {
+            VNRelinker handle;
+            const int actual_width = nodep->bitp()->width();  // Get actual bit width
+            std::string fmt;
+            // Normalize to standard bit width
+            if (actual_width <= 8) {
+                fmt = "#x%2x";
+            } else if (actual_width <= 16) {
+                fmt = "#x%4x";
+            } else if (actual_width <= 32) {
+                fmt = "#x%8x";
+            } else if (actual_width <= 64) {
+                fmt = "#x%16x";
+            } else {
+                nodep->v3warn(
+                    CONSTRAINTIGN,
+                    "Unsupported: Verilator does not currently support associative array index "
```

Comment:
```suggestion
                    "Unsupported: Associative array index "
```

And we need a test to fire this warning, along with a test that turns it off (to make sure your code here works with it disabled, which I suspect it doesn't.)